FW/Presentations/Lab_Guides/_C_01_Build_Setup_Download_EDK_II_Win_LabGuide.md

```diff
@@ -0,0 +1,350 @@
+\#Edk2 - main core code
+```
+$ git clone -b Edk2Lab_22Q3 https://github.com/tianocore-training/edk2.git
+```
+\#EDK II "C" Library Repo
+```
+$ git clone https://github.com/tianocore/edk2-libc.git
+```
+\#EDK II Platforms Repo
+```
+$ git clone https:://github.com/tianocore/edk2-platforms.git
+```
+\#EDK II Non-OSI (Stand alone Binaries)
+```
+$ git clone https:://github.com/tianocore/edk2-non-osi.git
+```
+\#Intel FSP
+```
+$ git clone https://github.com/IntelFsp/FSP.git
+```
+
+---
+## Slide 14 Download the EDK II Source Code
+### Download the open source EDK II from GitHub
+
+Download the Submodules and Checkout the Lab Branch
+```
+$ C:\fw\edk2-ws> cd edk2
+$ C:\fw\edk2-ws\edk2> git submodule update --init
+$ C:\fw\edk2-ws> cd ..
+```
+
+Download Checkout the Sha tag for edk2-platforms repo
+```
+$ C:\fw\edk2-ws> cd edk2-platforms
+$ C:\fw\edk2-ws> git reset --hard c546cc01f1517b42470f3ae44d67efcb8ee257fc
+```
+
+(rest to this commit since this is used with all the labs)
+
+---
+## Slide 15  Download Lab_Material_FW - getting the Source 
+Download the Lab_Material_FW.zip from: [github.com Lab_Matrial_FW.zip](https://github.com/tianocore-training/Lab_Material_FW/archive/refs/heads/main.zip)
+OR
+Use `git clone` to download the Lab_Material_FW
+```bash
+C:\> git clone https://github.com/tianocore-training/Lab_Material_FW.git
+```
+Directory Lab_Material_FW will be created
+
+```
+   FW 
+    - Documentation 
+   - DriverWizard 
+   - edk2-ws      
+   - edk2Linux 
+   - LabSampleCode 
+   - Nasm
+```
+
+---
+## Slide 16 Build EDK II
+### Extract the Lab Material
+
+Extract the Downloaded Lab_Material_FW-main.zip to C:\
+
+
+Download the Lab_Material_FW.zip from 
+
+
+---
+## Slide 17  Build  Edk2
+### Copy edk2-ws
+
+
+From the downloaded Lab_Material_FW folder **copy** and **paste** folder `"..\edk2-ws"` to `"C:/FW"`
+
+
+*Note:* Overwrite existing files and directories
+
+
+---
+## Slide 18  Build  Edk2
+### Get Nasm
+
+Copy `Nasm` directory to `C:\`
+
+(creating `C:\Nasm` directory)
+
+
+---
+## Slide 19  Build EDK II
+### Get IASL ACPI Compiler
+
+Copy `asl` directory to `C:\`
+
+(creating `C:\asl` directory)
+
+
+---
+## Slide 20 Build BaseTools
+
+Note: will need to update conf/target.txt for other labs
+
+
+---
+## Slide 21 Build EDK II
+### build BaseTools
+
+Open VS Command prompt & cd to workspace directory
+```
+$> cd C:\fw\edk2-ws
+```
+
+Setup the local environment: (see batch file `setenv.bat`)
+Sets WORKSPACE and PACKAGES_PATH env variables
+```
+$> setenv.bat
+```
+
+Invoke Edksetup.bat from directory C:/fw/edk2-ws/edk2 to Build BaseTools
+```
+$> cd edk2
+$> edksetup.bat Rebuild
+```
+
+**Important NOTE:** only use "Rebuild" on the first time - there after just use "edkseup.bat"
+
+
+---
+## Slide 22 Update Conf/Target.txt
+
+**Invoke** Edksetup.bat
+
+```
+$> cd C:\fw\edk2-ws\edk2
+$> edksetup.bat
+```
+
+**Edit** the file Conf/target.txt (**change** TOOL_CHAIN_TAG)
+```
+$> notepad Conf/target.txt
+```
+
```

Comment:
Change TOOL_CHAIN_TAG to

VS 2015 to VS2015x86
VS 2017 to VS2017
VS 2019 to VS2019

Change TARGET_ARCH to X64





<meta name=ProgId content=PowerPoint.Slide>
<meta name=Generator content="Microsoft PowerPoint 15">
<style>
<!--tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
td
	{padding-top:1.0px;
	padding-right:1.0px;
	padding-left:1.0px;
	mso-ignore:padding;
	color:windowtext;
	font-size:18.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Arial;
	mso-generic-font-family:auto;
	mso-font-charset:0;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;}
.oa1
	{border-top:1.0pt solid white;
	border-right:1.0pt solid white;
	border-bottom:3.0pt solid white;
	border-left:1.0pt solid white;
	background:#0071C5;
	mso-pattern:auto none;
	vertical-align:top;
	padding-bottom:3.6pt;
	padding-left:7.2pt;
	padding-top:3.6pt;
	padding-right:7.2pt;}
.oa2
	{border-top:3.0pt solid white;
	border-right:1.0pt solid white;
	border-bottom:1.0pt solid white;
	border-left:1.0pt solid white;
	background:#CBD5EA;
	mso-pattern:auto none;
	vertical-align:top;
	padding-bottom:3.6pt;
	padding-left:7.2pt;
	padding-top:3.6pt;
	padding-right:7.2pt;}
.oa3
	{border:1.0pt solid white;
	background:#E7EBF5;
	mso-pattern:auto none;
	vertical-align:top;
	padding-bottom:3.6pt;
	padding-left:7.2pt;
	padding-top:3.6pt;
	padding-right:7.2pt;}
-->
</style>
</head>

<body>
<!--StartFragment-->



2015 | VS2015x86
-- | --
2017 | VS2017
2019 | VS2019



<!--EndFragment-->
</body>

</html>